crates/matrix-sdk/src/matrix_auth/mod.rs

```diff
@@ -887,6 +887,24 @@ impl From<&login::v3::Response> for MatrixSession {
     }
 }
 
+impl TryFrom<&register::v3::Response> for MatrixSession {
+    type Error = ();
+    fn try_from(response: &register::v3::Response) -> Result<Self, ()> {
```

Comment:
One remark, I don't like that this is part of the public API. I think it should be a named constructor on `MatrixSession` (something like `from_register_response` returning `Option<Self>`). It might also make more sense to take the response by value instead of by reference, but I'm less certain about that.

Reply:
sure. if that is the way we want to go, I am happy to make that change.